app/src/sharedTest/java/org/oppia/app/parser/StringToRatioParserTest.kt

```diff
@@ -0,0 +1,104 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.RatioExpression
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [StringToRatioParser]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class StringToRatioParserTest {
+
+  private lateinit var stringToRatioParser: StringToRatioParser
+  private lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    context = ApplicationProvider.getApplicationContext()
+    stringToRatioParser = StringToRatioParser()
+  }
+
+  @Test
+  fun testParser_realtimeError_returnInvalidCharsError() {
```

Comment:
You should specify the condition being tested, eg: answerWithoutColons. 

Ditto for all other new tests in this PR.

Reply:
done